src/fallback.rs

```diff
@@ -32,16 +32,24 @@ pub static FISH_EMOJI_WIDTH: AtomicIsize = AtomicIsize::new(1);
 static WC_LOOKUP_TABLE: Lazy<WcLookupTable> = Lazy::new(WcLookupTable::new);
 
 /// A safe wrapper around the system `wcwidth()` function
+#[cfg(not(target_os = "cygwin"))]
 pub fn wcwidth(c: char) -> isize {
     extern "C" {
         pub fn wcwidth(c: libc::wchar_t) -> libc::c_int;
     }
 
-    const _: () = assert!(mem::size_of::<libc::wchar_t>() >= mem::size_of::<char>());
+    const _: () = assert!(std::mem::size_of::<libc::wchar_t>() >= std::mem::size_of::<char>());
     let width = unsafe { wcwidth(c as libc::wchar_t) };
     isize::try_from(width).unwrap()
 }
 
+/// A safe wrapper around the system `wcwidth()` function
+#[cfg(target_os = "cygwin")]
+pub fn wcwidth(c: char) -> isize {
+    // https://github.com/fish-shell/fish-shell/pull/11238#discussion_r1985414670
+    2
```

Comment:
You want a hardcoded width of 2 for `😃`, which is our test codepoint to see if the system knows Unicode 9.

You *don't* want this to return 2 for everything.

I think this function should not be compiled on cygwin at all - instead the 2 should be hardcoded in https://github.com/fish-shell/fish-shell/blob/90218821f9a373684b439abe5cf200dc467fa7fe/src/env_dispatch.rs#L200-L204

Reply:
And https://github.com/fish-shell/fish-shell/blob/90218821f9a373684b439abe5cf200dc467fa7fe/src/fallback.rs#L51-L53

should be stubbed out - cygwin doesn't have a "console session".